### ISSUE TITLE
System.Linq needs to support 4.1.0.0 on desktop or increase the generation

### ISSUE THREAD

**Issue by ericstj** [dotnet/corefx#6017]
I've added a baseline for our internal build to permit the missing members for now, but you either need to add these APIs to desktop or create an API surface that is OOB-able to desktop.  An example of an OOB-able design would be to put the new members on a new type.

If you can OOB the new APIs then you can make 4.1.0 dotnet5.4/netstandard1.3.  It will work on .NET 4.6 and later.
If you cannot OOB the new APIs (or do not want to ) and you add the APIs to .NET 4.6.2/4.7 then you can make 4.0 dotnet5.6/netstandard1.5.
If you cannot OOB the new APIs and you cannot add the APIs to .NET 4.6.2/4.7 then we must create a new generation and put the APIs there: dotnet5.7/netstandard1.6.  Future versions of desktop must implement the new API in order to claim support for the new generation.

/cc @weshaggard @stephentoub

**Comment by VSadov** [comment]
I'd think adding the new API to desktop is a better option.

NOTE: same applies to the Expression.Compile(preferinterpretation) overload. It would be ok to just add a method, but still defer to compiling, since it is "prefer" setting.

**Comment by weshaggard** [comment]
Yes I think we should just add these to .NET 4.7. @VSadov can you please make sure you have bugs tracking the addition of these?

**Comment by NickCraver** [comment]
How will new generations be tracked in general? e.g. "Here's the list of new surface area of 1.6" Apologies if this exists and I missed it, but I haven't found any info on early thoughts on how to aggregate this information. 

If there's no plan yet: might I suggest a single issue isn't the worst idea given the current ecosystem. Like [C#7 feature tracking in #2136](https://github.com/dotnet/roslyn/issues/2136); does something like this or a plan already exist for future generations? I know it's a bit early to really get any realistic picture of the next generation, but .Net core has certainly shown the deficiencies in desktop we're designing around that won't be solved until another generation. Given that current designs already have future plans for refinement in later generations - can we start tracking that somewhere?

**Comment by weshaggard** [comment]
@NickCraver That is a fair question one which I don't actually even fully know how to track yet myself. As the plans begin to finalize around netstandard and generations I expect we will begin to get a clearer picture about how this all fits together and @ericstj plans to provide some more documentation on this area. However what I do know is that I don't think we will be able to easily track API additions for a new generation in an issue like the C# 7.0 features as every library will need to be handled differently as they will all be in different generations based on their dependencies and whether or not they themselves ship frozen inbox or can ship out-of-band.

**Comment by venkat-raman251** [comment]
Remove ApiCompatBaseline.net462.txt once errors mentioned in it are resolved. @weshaggard

**Comment by weshaggard** [comment]
@VSadov this needs to be fixed before 1.0.0-rtm.

```
MembersMustExist : Member 'System.Linq.Enumerable.Append<TSource>(System.Collections.Generic.IEnumerable<TSource>, TSource)' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'System.Linq.Enumerable.Prepend<TSource>(System.Collections.Generic.IEnumerable<TSource>, TSource)' does not exist in the implementation but it does exist in the contract.
```
The above APICompat issues represent real erros and so we need to fix this before we ship this System.Linq package as stable otherwise people using these API will not actually work on desktop.
